### ISSUE TITLE
Regression: deconstructed parameters not supported

### ISSUE THREAD

**Issue by sneakypete81** [gajus/eslint-plugin-jsdoc#352]
This regression was introduced in v15.5.4.

### Steps to Reproduce
Run eslint on the following example adapted from https://stackoverflow.com/a/45788619:
```javascript
/**
 * Assign the project to an employee.
 *
 * @param {object} employee - The employee who is responsible for the project.
 * @param {string} employee.name - The name of the employee.
 * @param {string} employee.department - The employee's department.
 */
export function assign({name, department}) {
  // ...
}
```

### Expected Result (v15.5.3)
Pass without error.
### Actual Result (v15.5.4)
```
filename.js 1:1  error  Missing JSDoc @param "<ObjectPattern>" declaration  jsdoc/require-param
```

**Comment by sneakypete81** [comment]
Thanks, that's fixed it!
